owners/test/owners.test.js

```diff
@@ -254,9 +258,77 @@ describe('owners rules', () => {
   describe('basic directory ownership', () => {
     describe('matchesFile', () => {
       it('matches all files', () => {
-        expect('src/OWNERS.yaml').toMatchFile('src/foo.txt');
-        expect('src/OWNERS.yaml').toMatchFile('src/foo/bar.txt');
-        expect('src/OWNERS.yaml').toMatchFile('src/foo/bar/baz.txt');
+        const rule = new OwnersRule('src/OWNERS.yaml', []);
+
+        expect(rule).toMatchFile('src/foo.txt');
+        expect(rule).toMatchFile('src/foo/bar.txt');
+        expect(rule).toMatchFile('src/foo/bar/baz.txt');
+      });
+    });
+  });
+
+  describe('with glob patterns', () => {
+    describe('constructor', () => {
+      it.each([
+        ['file.txt', /file\.txt/],
+        ['package*.json', /package.*?\.json/],
+        ['*.css, *.js', /.*?\.css|.*?\.js/],
+      ])(
+        'converts the pattern "%p" into regex "%p"',
+        (pattern, expectedRegex) => {
+          const rule = new PatternOwnersRule('OWNERS.yaml', [], pattern);
+          expect(rule.regex).toEqual(expectedRegex);
+        }
+      );
+    });
+
+    describe('regexEscape', () => {
+      it.each([
+        ['file.txt', 'file\\.txt'],
+        ['this+that.txt', 'this\\+that\\.txt'],
+        ['*.css, *.js', '\\*\\.css, \\*\\.js'],
+        ['with space.txt', 'with space\\.txt'],
+        ['with-hyphen.txt', 'with-hyphen\\.txt'],
+      ])('escapes "%p" as "%p"', (text, expected) => {
+        expect(PatternOwnersRule.escapeRegexChars(text)).toEqual(expected);
```

Comment:
This seems like it should only be an internal method. Can you test the higher-level function that calls this instead, like you do above?

Reply:
Sure, I'll move some of these tests to `matchesFile`